core/Columns/Dimension.php

```diff
@@ -167,6 +170,24 @@ abstract class Dimension
      */
     protected $sqlFilterValue;
 
+    /**
+     * A callback that can be used to customize the SQL used to match a segment. The value must be a callable
+     * with the following signature:
+     *
+     * ```
+     * function ($field, $matchType, $value, $join) {
+     *    // ...
+     * }
+     * ```
+     *
+     * Return `null` to use the default SQL. Otherwise, return an array with two keys: `sql` and `bind`
+     * where `sql` is a SQL string and `bind` is an array of values to bind when executing it.
+     *
+     * @var callable
+     * @api since Piwik 5.0.0
+     */
+    protected $sqlFilterMatch;
```

Comment:
What exactly is this for? Looking through the chances it looks like this new property has been added for dimensions and segments, but there is not even once dimension or segment that already uses this property. Is this on purpose and maybe needed for some of our plugins? 🤔

Reply:
Originally this was to convert conditions like `pageUrl==abc` to `log_action.name = ? AND log_action.hash = CRC32(?) AND log_action.type = 1` instead of just `log_action.name = ?`. This was changed to not use a join for `==` operators, but @tsteur decided to sqlFilterMatch was worth keeping in: https://github.com/matomo-org/matomo/pull/20379#issuecomment-1461271603

Reply:
So this adds a "feature", that is not actually used at all and does not have any tests. Without having that commented and well documented somewhere I guess it's almost worthless to have it, as no one will know how and when to use it.

Reply:
Maybe @tsteur has something to add, or do you want me to add tests?

Reply:
@sgiehl @diosmosis the usage of `sqlFilterMatch` in https://github.com/matomo-org/matomo/pull/20379/commits/dc48518ccc50d81e67c20cecdcc230f35aec4449 was actually removed after my comment. When I had a look there was still some good usage of that to improve the speed of those queries which is critical to have. we could remove it if there's no usage of it anymore.

Reply:
@tsteur to be clear would you **rather** it be removed if there is no current usage?

Reply:
I think that be good @diosmosis

Reply:
@sgiehl updated